src/main/java/com/android/volley/toolbox/CacheHeader.java

```diff
@@ -115,6 +120,44 @@ static CacheHeader readHeader(DiskBasedCache.CountingInputStream is) throws IOEx
                 key, etag, serverDate, lastModified, ttl, softTtl, allResponseHeaders);
     }
 
+    /**
+     * Reads the header from a AsynchronousFileChannel and returns a CacheHeader object.
+     *
+     * @param afc File channel to read from.
+     * @param size Size of the cache header.
+     * @throws IOException if fails to read header
+     */
+    @RequiresApi(api = Build.VERSION_CODES.O)
+    static CacheHeader readHeader(final AsynchronousFileChannel afc, int size) throws IOException {
+        final ByteBuffer buffer = ByteBuffer.allocate(size);
+        Future<Integer> fillBuffer = afc.read(buffer, 0);
+        try {
+            fillBuffer.get();
+        } catch (ExecutionException e) {
+            VolleyLog.e(e.toString(), "ExecutionException thrown");
+            throw new IOException(e);
+        } catch (InterruptedException e) {
+            VolleyLog.e(e.toString(), "InterruptedException thrown");
```

Comment:
`caught` not `thrown`?

I think this is a reasonable way to handle interruption.